### ISSUE TITLE
android.os.BadParcelableException in cgeo.geocaching.maps.routing

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#8226]
Reported as new crash with 2020.04.15-RC on Google Play.
Several devices running Android 9 or 10.

```
java.lang.RuntimeException: 
 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3488)
 
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3635)
 
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:83)
 
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
 
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
 
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2175)
 
  at android.os.Handler.dispatchMessage (Handler.java:107)
 
  at android.os.Looper.loop (Looper.java:237)
 
  at android.app.ActivityThread.main (ActivityThread.java:7857)
 
  at java.lang.reflect.Method.invoke (Native Method)
 
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1076)
Caused by: android.os.BadParcelableException: 
 
  at android.os.Parcel.readParcelableCreator (Parcel.java:3059)
 
  at android.os.Parcel.readParcelable (Parcel.java:2981)
 
  at android.os.Parcel.readValue (Parcel.java:2883)
 
  at android.os.Parcel.readListInternal (Parcel.java:3324)
 
  at android.os.Parcel.readArrayList (Parcel.java:2467)
 
  at cgeo.geocaching.maps.routing.Route.<init> (Route.java:261)
 
  at cgeo.geocaching.maps.routing.Route$1.createFromParcel (Route.java:250)
 
  at cgeo.geocaching.maps.routing.Route$1.createFromParcel (Route.java:246)
 
  at android.os.Parcel.readParcelable (Parcel.java:2990)
 
  at android.os.Parcel.readValue (Parcel.java:2883)
 
  at android.os.Parcel.readArrayMapInternal (Parcel.java:3261)
 
  at android.os.BaseBundle.initializeFromParcelLocked (BaseBundle.java:292)
 
  at android.os.BaseBundle.unparcel (BaseBundle.java:236)
 
  at android.os.BaseBundle.getBoolean (BaseBundle.java:899)
 
  at android.app.Activity.restoreHasCurrentPermissionRequest (Activity.java:8279)
 
  at android.app.Activity.performCreate (Activity.java:7951)
 
  at android.app.Activity.performCreate (Activity.java:7944)
 
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1307)
 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3463)
```

**Comment by Lineflyer** [comment]
Another similar exception on an Android 8 device:

```
java.lang.RuntimeException: 
 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2825)
 
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2900)
 
  at android.app.ActivityThread.-wrap11 (Unknown Source)
 
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1601)
 
  at android.os.Handler.dispatchMessage (Handler.java:105)
 
  at android.os.Looper.loop (Looper.java:251)
 
  at android.app.ActivityThread.main (ActivityThread.java:6589)
 
  at java.lang.reflect.Method.invoke (Native Method)
 
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:240)
 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:767)
Caused by: android.os.BadParcelableException: 
 
  at android.os.Parcel.readParcelableCreator (Parcel.java:2889)
 
  at android.os.Parcel.readParcelable (Parcel.java:2815)
 
  at android.os.Parcel.readValue (Parcel.java:2718)
 
  at android.os.Parcel.readListInternal (Parcel.java:3146)
 
  at android.os.Parcel.readArrayList (Parcel.java:2348)
 
  at cgeo.geocaching.maps.routing.Route.<init> (Route.java:261)
 
  at cgeo.geocaching.maps.routing.Route$1.createFromParcel (Route.java:250)
 
  at cgeo.geocaching.maps.routing.Route$1.createFromParcel (Route.java:246)
 
  at android.os.Parcel.readParcelable (Parcel.java:2824)
 
  at android.os.Parcel.readValue (Parcel.java:2718)
 
  at android.os.Parcel.readArrayMapInternal (Parcel.java:3085)
 
  at android.os.BaseBundle.unparcel (BaseBundle.java:257)
 
  at android.os.BaseBundle.getBoolean (BaseBundle.java:825)
 
  at android.app.Activity.restoreHasCurrentPermissionRequest (Activity.java:7223)
 
  at android.app.Activity.performCreate (Activity.java:6974)
 
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1214)
 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2778)
```

**Comment by moving-bits** [comment]
Hm, haven't been able to reproduce this error so far, which makes it somewhat hard to fix.

There are some SO links for that message, but the "accepted answers" suggest changes that only lead to the route functions no longer working at all anymore...

Do we have some more info about what the users have been doing before this error? And if it's reproducible?

**Comment by Lineflyer** [comment]
#8231 is merged. Lets see.

**Comment by moving-bits** [comment]
Closed with #8231
